firmware/src/gui/pages/manual_control_popup.hh

```diff
@@ -194,10 +194,45 @@ private:
 		float range = lv_arc_get_max_value(ui_ControlArc) - lv_arc_get_min_value(ui_ControlArc);
 		auto value = lv_arc_get_value(ui_ControlArc) - lv_arc_get_min_value(ui_ControlArc);
 
+		float normalized_value = (float)value / range;
+
+		// Check if this is a snapped knob so that we can just jump to the nearest value
+		std::visit([&](const auto& el) {
+			using T = std::decay_t<decltype(el)>;
+			if constexpr (std::is_base_of_v<Pot, T>) {
```

Comment:
I think using `is_base_of` is OK, though everywhere else in the firmware code, we use the `overloaded` helper class, so we should follow suit here. It also makes it easier to add more cases for other element types later. See lines 164-174 above, for an example.